data-pipeline-flink/dp-core/src/main/scala/org/sunbird/dp/core/job/BaseJobConfig.scala

```diff
@@ -21,6 +21,7 @@ class BaseJobConfig(val config: Config, val jobName: String) extends Serializabl
   val kafkaProducerMaxRequestSize: Int = config.getInt("kafka.producer.max-request-size")
   val kafkaProducerBatchSize: Int = config.getInt("kafka.producer.batch.size")
   val kafkaProducerLingerMs: Int = config.getInt("kafka.producer.linger.ms")
+  val kafkaProducerCompression: String = config.getString("kafka.producer.compression")
```

Comment:
We will need the default compression to be set to snappy. The cpu utilization with snappy is definitely lower and hence we have defaulted to snappy.

Reply:
On NIC systems snappy library is not installed by the setup process, since it has to access some external url to do this access to which is restricted on their systems. Flink on the other hand runs in Kubernetes pods, which have no such issues with the snappy lib. This results in flink jobs producing messages that are compressed using snappy, but other systems running secor processes and druid ingestion, fail to read these messages because of snappy lib not being present.

Setting compression to none is a quick workaround to this issue. This change keeps the default compression as snappy, but provides an option to configure producer compression as none (or any other option).

Let me know if this is not a desirable change, will remove if that is the case, but we'll need to figure out a way to fix the snappy lib issue on NIC servers.

Reply:
I understand that the compression type should be configurable. However, we can provide the default value for the configurartion as `snappy`. This allows that the projects which have not defined this configuration to `snappy` and will allow NIC to configure the compression type to `None`.

Reply:
I have added snappy as the default, the code snippet above is outdated